src/lib/mina_lib/mina_lib.ml

```diff
@@ -901,6 +901,13 @@ let add_work t (work : Snark_worker_lib.Work.Result.t) =
     (Network_pool.Snark_pool.Resource_pool.Diff.of_result work, cb)
   |> Deferred.don't_wait_for
 
+let add_work_graphql t diff =
+  let results_ivar = Ivar.create () in
```

Comment:
I'm not sure a an Ivar push makes sense as a validation call back here.

Reply:
I thought the purpose of this callback was to check the validation status and handle it appropriately for a given message.

Reply:
> Is an ivar necessary here? Can we not just bind the deferred value and return?
Yes, if you want to return the status of the operation- whether the completed work was broadcasted or not.

>I thought the purpose of this callback was to check the validation status and handle it appropriately for a given message.
Not sure what the discrepancy is here. The validation status is returned to the grapqhl request and as for the message, it would either be added to the pool or not